test/uq_parameters.jl

```diff
@@ -0,0 +1,224 @@
+    # should return
+    target_map = Dict(
+        "uq_param_1" => ParameterDistribution(
+            Parameterized(Normal(2.0, 1.0)),
+            no_constraint(),
+            "uq_param_1"),
+
+        "uq_param_2" => ParameterDistribution(
+            Parameterized(Gamma(5.0, 2.0)),
+            bounded_below(3.0),
+            "uq_param_2"),
+
+        "uq_param_3" => ParameterDistribution(
+            Parameterized(MvNormal(4, 0.1)),
+            [no_constraint(), bounded_below(-1.0),
+            bounded_above(0.4), bounded(-0.1, 0.2)],
+            "uq_param_3"),
+
+        "uq_param_4" => ParameterDistribution(
+            Samples([1.0 3.0 5.0 7.0; 9.0 11.0 13.0 15.0]),
+            [bounded(10.0, 15.0), bounded_below(-1.0)],
+            "uq_param_4"),
+
+        "uq_param_5" => ParameterDistribution(
+            Samples([1.0 3.0; 5.0 7.0; 9.0 11.0; 13.0 15.0]),
+            [no_constraint(), no_constraint(),
+             bounded_below(-2.0), bounded_above(100.0)],
+            "uq_param_5"),
+
+        "uq_param_6" => ParameterDistribution(
+            [Parameterized(Gamma(2.0, 1.0)), Parameterized(Gamma(2.0, 1.0)),
+             Parameterized(Gamma(2.0, 1.0))],
+            [[bounded_above(9.0)], [bounded_above(9.0)], [bounded_above(9.0)]],
+            ["uq_param_6_(1)", "uq_param_6_(2)", "uq_param_6_(3)"]),
+
+        "uq_param_7" => ParameterDistribution(
+            Parameterized(MvNormal(3, 2.0)),
+            [no_constraint(), no_constraint(), no_constraint()],
+            "uq_param_7")
+    )
+
+    # Get all `ParameterDistribution`s. We also add dummy (key, value) pairs
+    # to check if that information gets added correctly when saving the
+    # parameters back to file and re-loading them
+    descr = " will be learned using CES"
+    for param_name in keys(param_dict)
+        param_dict[param_name]["description"] = param_name * descr
+        pd = CP.get_parameter_distribution(param_dict, param_name)
+        target_pd = target_map[param_name]
+
+        # Check names
+        @test get_name(pd) == get_name(target_pd)
+        # Check distributions
+        @test get_distribution(pd) == get_distribution(target_pd)
+        # Check constraints
+        constraints = get_all_constraints(pd)
+        target_constraints = get_all_constraints(target_pd)
+        @test constraints == target_constraints
+
+    end
+
+    # We can also get a `ParameterDistribution` representing
+    # multiple parameters
+    param_names = ["uq_param_2", "uq_param_4", "uq_param_5"]
+    pd = CP.get_parameter_distribution(param_dict, param_names)
+
+    # Save the parameter dictionary and re-load it. 
+    logfile_path = joinpath(@__DIR__,"log_file_test_uq.toml")
+    CP.write_log_file(param_dict, logfile_path)
+
+    # Read in log file as new parameter file and rerun test.
+    param_dict_from_log = CP.parse_toml_file(logfile_path)
+    rm(logfile_path) # clean up
```

Comment:
move this below the test in case it fails partway through, or even leave it as it is a log-file. In general these will overwrite existing logs I believe.

Reply:
I removed the remove command